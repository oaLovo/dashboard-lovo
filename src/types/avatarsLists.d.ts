export type ListDataProps = {
  avatar?: string
  name: string
  subtitle: string
}

export interface AvatarsListsProps {
  listData: ListDataProps[]
}
