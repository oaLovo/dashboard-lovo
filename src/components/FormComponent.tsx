import styled from 'styled-components'
import { FormComponentProps } from '@/types'
import { StyledButton, StyledInput, StyledP } from '@/components'
import { pxToRem } from '@/utils'

const StyledForm = styled.form`
  display: flex;
  flex-direction: column;
  row-gap: ${pxToRem(16)};
`

function FormComponent(props: FormComponentProps) {
  const { inputs, buttons, message } = props
  return (
    <StyledForm>
      {inputs.map((inputProps, index) => (
        <StyledInput key={index} {...inputProps} />
      ))}
      {buttons.map((buttonProps, index) => (
        <StyledButton key={index} {...buttonProps} />
      ))}
      {message && (
        <StyledP className={message.type === 'error' ? 'error' : 'success'}>
          {message.msg}
        </StyledP>
      )}
    </StyledForm>
  )
}

export default FormComponent
