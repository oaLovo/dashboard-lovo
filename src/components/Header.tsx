import styled from 'styled-components'
import { Link } from 'react-router-dom'

import { Logo } from '@/components'

// MUI
import Avatar from '@mui/material/Avatar'
import Box from '@mui/material/Box'
import Container from '@mui/material/Container'

// UTILS
import { pxToRem } from '@/utils'

const StyledHeader = styled.header`
  background-color: ${(props) => props.theme.appBackground};
  border-bottom: ${pxToRem(1)} solid ${(props) => props.theme.appDefaultStroke};
  margin-bottom: ${pxToRem(37)};
  width: 100%;
`

function Header() {
  return (
    <StyledHeader>
      <Container maxWidth="lg">
        <Box
          sx={{
            alignItems: 'center',
            display: 'flex',
            justifyContent: 'space-between',
            height: pxToRem(64),
          }}
        >
          <Link to="/home">
            <Logo height={30} width={73} />
          </Link>
          <Link to="/perfil">
            <Avatar
              alt="DNC Avatar"
              src="/dnc-avatar.jpg"
              sx={{ width: pxToRem(40), height: pxToRem(40) }}
            />
          </Link>
        </Box>
      </Container>
    </StyledHeader>
  )
}

export default Header
