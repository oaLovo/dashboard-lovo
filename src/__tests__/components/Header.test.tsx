import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import { BrowserRouter as Router } from 'react-router-dom'
import { Header } from '@/components'
import { themesList } from '@/resources/themesList'
import { Theme } from '@/types'

describe('Header Component', () => {
  const renderComponent = (theme: Theme) =>
    render(
      <ThemeProvider theme={theme}>
        <Router>
          <Header />
        </Router>
      </ThemeProvider>
    )

  themesList.forEach(({ name, theme }) => {
    describe(`${name}`, () => {
      it('should match the snapshot', () => {
        const { container } = renderComponent(theme)
        expect(container).toMatchSnapshot()
      })
    })
  })
})
