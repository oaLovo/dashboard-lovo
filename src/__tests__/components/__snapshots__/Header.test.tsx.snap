// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Dark Theme should match the snapshot 1`] = `
.c0 {
  background-color: #060B26;
  border-bottom: 0.0625rem solid #21497D;
  margin-bottom: 2.3125rem;
  width: 100%;
}

.c1 {
  background-image: url(/dnc-logo-white.svg);
  background-size: cover;
  height: 1.875rem;
  width: 4.5625rem;
}

<div>
  <header
    class="c0"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
    >
      <div
        class="MuiBox-root css-alor41"
      >
        <a
          href="/home"
        >
          <figure
            class="c1"
            height="30"
            width="73"
          />
        </a>
        <a
          href="/perfil"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular css-rk6u8x-MuiAvatar-root"
          >
            <img
              alt="DNC Avatar"
              class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
              src="/dnc-avatar.jpg"
            />
          </div>
        </a>
      </div>
    </div>
  </header>
</div>
`;

exports[`Header Component Light Theme should match the snapshot 1`] = `
.c0 {
  background-color: #FFF;
  border-bottom: 0.0625rem solid #E0E0E0;
  margin-bottom: 2.3125rem;
  width: 100%;
}

.c1 {
  background-image: url(/dnc-logo-black.svg);
  background-size: cover;
  height: 1.875rem;
  width: 4.5625rem;
}

<div>
  <header
    class="c0"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
    >
      <div
        class="MuiBox-root css-alor41"
      >
        <a
          href="/home"
        >
          <figure
            class="c1"
            height="30"
            width="73"
          />
        </a>
        <a
          href="/perfil"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular css-rk6u8x-MuiAvatar-root"
          >
            <img
              alt="DNC Avatar"
              class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
              src="/dnc-avatar.jpg"
            />
          </div>
        </a>
      </div>
    </div>
  </header>
</div>
`;
